### PR TITLE
Creates a tube class

### PR DESCRIPTION
**Pull request by rolfvdhulst**
Represents a tube, which is e.g. the shape of the illegal ball area or the obstacle of a linear object (like a wall of the goal).

### CODE REVIEW

File: include/roboteam_utils/Tube.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+         * @param radius Given radius
+         */
+        Tube(const Vector2& start, const Vector2& end, double radius);
+        /**
+         * @brief Construct a tube around the line with given radius
+         * @param line Line to construct tube around
+         * @param radius Given radius
+         */
+        Tube(const LineSegment& line, double radius);
+
```
Comment (emielsteerneman) [inline include/roboteam_utils/Tube.h]: Can we construct a tube by passing another tube?

File: include/roboteam_utils/Tube.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+         * @return True if point is inside or on the boundary of the tube
+         */
+        [[nodiscard]] bool contains(const Vector2 &point) const;
+        /**
+         * @brief checks if the line intersects the tube or is contained within the tube
+         * @param line to consider
+         * @return True if the line intersects with the tube
+         */
+        [[nodiscard]] bool doesIntersectOrContain(const LineSegment &line) const;
+
```
Comment (emielsteerneman) [inline include/roboteam_utils/Tube.h]: Intersections with a circle might be useful as well, because are basically circles

Comment (rolfvdhulst) [reply include/roboteam_utils/Tube.h]: Ok, can easily implement this (it's literally a one liner)

File: include/roboteam_utils/Tube.h (hunk 1 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/Tube.h]: Yes, but this should be defaulted correctly already. We can be explicit in the header but it won´t make an effective difference.

**Pull request merged**